src/main/java/com/mixpanel/android/mpmetrics/DecideMessages.java

```diff
@@ -175,6 +175,12 @@ public synchronized InAppNotification getNotification(int id, boolean replace) {
         return notif;
     }
 
+    // if a notification was failed to show, add it back to the unseen list so that we
+    // won't lose it
+    public synchronized void handleNotificationDisplayFailure(InAppNotification notif) {
```

Comment:
For now we could just change the name of this method to something like `markNotificationAsUnseen`